source/extensions/filters/network/mysql_proxy/mysql_codec_switch_resp.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/network/mysql_proxy/mysql_codec.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MySQLProxy {
+
+class ClientSwitchResponse : public MySQLCodec {
+public:
+  // MySQLCodec
+  int parseMessage(Buffer::Instance& buffer, uint64_t& offset, int len) override;
+  std::string encode() override;
+
+  const std::string& getAuthPluginResp() const { return auth_plugin_resp_; }
```

Comment:
prefer non-abbreviation for filename, method name and member variable name. just use full "response"?